src/components/AnimateSharedLayout/SharedLayoutTree.tsx

```diff
@@ -0,0 +1,108 @@
+import * as React from "react"
+import {
+    SharedLayoutContext,
+    SharedLayoutSyncMethods,
+} from "./SharedLayoutContext"
+import { HTMLVisualElement } from "render/dom/HTMLVisualElement"
+import { Presence } from "./types"
+
+export interface SharedLayoutTreeProps {
+    /**
+     * @public
+     */
+    children: React.ReactNode
+
+    /**
+     * Are the children of the tree valid targets when computing lead and follow.
+     */
+    isPresent: boolean | undefined
+    /**
+     * Are the children of the tree valid targets when computing lead and follow.
+     */
+    isValidTree: boolean
+
+    /**
+     * What priority should children of this tree take when determining the lead and follow.
+     */
+    layoutOrder: number
+
+    isLead: boolean
+}
+
+interface SyncProps extends SharedLayoutTreeProps {
+    syncContext: SharedLayoutSyncMethods
+}
+
+export class LayoutTree extends React.Component<SyncProps> {
```

Comment:
I started with this extending AnimateSharedLayout but there were circular dependancies and it was getting more confusing dealign with the types than just breaking it off. Happy to reconsider if you had something in mind though!